arch/Kconfig

```diff
@@ -33,6 +33,11 @@ config RISCV32
 config XTENSA
 	bool "Xtensa architecture"
 
+config ARCH_POSIX
+	bool "POSIX (native) architecture"
+	select ATOMIC_OPERATIONS_C
```

Comment:
why not use ATOMIC_OPERATIONS_BUILTIN?

Reply:
ATOMIC_OPERATIONS_BUILTIN was my first proposal (in garbage/autoconf.h). 
@nashif any reason why you set it ATOMIC_OPERATIONS_C in Kconfig?